src/renderer/components/+preferences/preferences.tsx

```diff
@@ -18,279 +18,95 @@
-  Other = "other"
-}
+import { Application } from "./application";
+import { Kubernetes } from "./kubernetes";
+import { LensProxy } from "./proxy";
+import { Telemetry } from "./telemetry";
 
 @observer
 export class Preferences extends React.Component {
-  @observable httpProxy = UserStore.getInstance().httpsProxy || "";
-  @observable shell = UserStore.getInstance().shell || "";
-  @observable activeTab = Pages.Application;
+  @observable historyLength: number | undefined;
 
   constructor(props: {}) {
     super(props);
     makeObservable(this);
   }
 
-  @computed get themeOptions(): SelectOption<string>[] {
-    return ThemeStore.getInstance().themes.map(theme => ({
-      label: theme.name,
-      value: theme.id,
-    }));
-  }
-
-  timezoneOptions: SelectOption<string>[] = moment.tz.names().map(zone => ({
-    label: zone,
-    value: zone,
-  }));
-
-  componentDidMount() {
-    disposeOnUnmount(this, [
-      reaction(() => navigation.location.hash, hash => {
-        const fragment = hash.slice(1); // hash is /^(#\w.)?$/
-
-        if (fragment) {
-          // ignore empty fragments
-          document.getElementById(fragment)?.scrollIntoView();
-        }
-      }, {
-        fireImmediately: true
-      })
-    ]);
-  }
-
-  onTabChange = (tabId: Pages) => {
-    this.activeTab = tabId;
-  };
-
   renderNavigation() {
     const extensions = AppPreferenceRegistry.getInstance().getItems().filter(e => !e.showInPreferencesTab);
+    const currentLocation = navigation.location.pathname;
+    const isActive = (route: RouteProps) => !!matchPath(currentLocation, { path: route.path, exact: route.exact });
 
     return (
-      <Tabs className="flex column" scrollable={false} onChange={this.onTabChange} value={this.activeTab}>
+      <Tabs className="flex column" scrollable={false} onChange={(url) => navigateWithoutHistoryChange({ pathname: url })}>
         <div className="header">Preferences</div>
-        <Tab value={Pages.Application} label="Application" data-testid="application-tab"/>
-        <Tab value={Pages.Proxy} label="Proxy" data-testid="proxy-tab"/>
-        <Tab value={Pages.Kubernetes} label="Kubernetes" data-testid="kube-tab"/>
-        <Tab value={Pages.Telemetry} label="Telemetry" data-testid="telemetry-tab"/>
+        <Tab value={appURL()} label="Application" data-testid="application-tab" active={isActive(appRoute)}/>
+        <Tab value={proxyURL()} label="Proxy" data-testid="proxy-tab" active={isActive(proxyRoute)}/>
+        <Tab value={kubernetesURL()} label="Kubernetes" data-testid="kube-tab" active={isActive(kubernetesRoute)}/>
+        <Tab value={telemetryURL()} label="Telemetry" data-testid="telemetry-tab" active={isActive(telemetryRoute)}/>
         {extensions.length > 0 &&
-          <Tab value={Pages.Extensions} label="Extensions" data-testid="extensions-tab"/>
+          <Tab value={extensionURL()} label="Extensions" data-testid="extensions-tab" active={isActive(extensionRoute)}/>
         }
```

Comment:
I know you brought this up https://github.com/lensapp/lens/pull/3234#issuecomment-892552066 but I am still wondering where you see the controversy, or do you mean that you would prefer more control over how the settings pages are rendered?

Reply:
This is simple. Problems as I see in #3234
1. We have reduced the control over the styling of elements. In here https://github.com/lensapp/lens/pull/3234/files#diff-8111c23d78b73e3e7dac1c6d8a6060060af33487ccaa99d78326dd9865602d08R108 we iterate every "preference" items with predefined html elements: a title, a hint and a `<hr/>` elements. This looks quite nice, but in reality some sections may look different, some of them may not contain `<hr/>`, some may have more paddings etc. #3234 makes impossible to fine-tune such a things.
2. By moving all preference sections into "registry" we have violated the natural order of the display according to the React. Instead of placing an element inside JSX like so:
```
<section id='sentry' className="small">
  <SubTitle title='Automatic Error Reporting' />
  <Checkbox
    label="Allow automatic error reporting"
    value={UserStore.getInstance().allowErrorReporting}
    onChange={value => {
      UserStore.getInstance().allowErrorReporting = value;
    }}
 />
</section>
```
#3234 produces one level of abstraction to point React where to place elements:
```
{
  id: "sentry",
  title: "Automatic Error Reporting",
  components: {
          Input: () => (
            <Checkbox
              label="Allow automatic error reporting"
              value={UserStore.getInstance().allowErrorReporting}
              onChange={value => {
                UserStore.getInstance().allowErrorReporting = value;
              }}
            />
          ),
          Hint: () => (
            <>
              Automatic error reports provide vital information about issues and application crashes.{" "}
              It is highly recommended to keep this feature enabled to ensure fast turnaround for issues you might encounter.
            </>
          )
        },
        showInPreferencesTab: AppPreferenceKind.Telemetry,
        hide: Boolean(sentryDsn),
}
```